GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/DehydrateTests.cs

```diff
@@ -143,14 +163,363 @@ public void DehydrateShouldFailOnWrongDiskLayoutVersion()
+        public void FolderDehydrateRelativePaths()
+        {
+            string[] foldersToDehydrate = new[]
+            {
+                Path.Combine("..", ".gvfs"),
+                Path.DirectorySeparatorChar + Path.Combine("..", ".gvfs"),
+                Path.Combine("GVFS", "..", "..", ".gvfs"),
+                Path.Combine("GVFS/../../.gvfs"),
+            };
+
+            List<string> errorMessages = new List<string>();
+            foreach (string path in foldersToDehydrate)
+            {
+                errorMessages.Add($"Cannot dehydrate '{path}' folder.  Invalid folder path.");
+            }
+
+            this.DehydrateShouldSucceed(
+                errorMessages.ToArray(),
+                confirm: true,
+                noStatus: false,
+                foldersToDehydrate: foldersToDehydrate);
+        }
+
+        [TestCase]
+        public void FolderDehydrateFolderThatDoesNotExist()
+        {
+            this.DehydrateShouldSucceed(new[] { "DoesNotExist did not exist to dehydrate." }, confirm: true, noStatus: false, foldersToDehydrate: "DoesNotExist");
+        }
+
+        [TestCase]
+        public void FolderDehydrateNewlyCreatedFolderAndFile()
+        {
+            string directoryToCreate = this.Enlistment.GetVirtualPathTo("NewFolder");
+            this.fileSystem.CreateDirectory(directoryToCreate);
+            string fileToCreate = Path.Combine(directoryToCreate, "newfile.txt");
+            this.fileSystem.WriteAllText(fileToCreate, "Test content");
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "add .");
+            GitProcess.Invoke(this.Enlistment.RepoRoot, "commit -m Test");
+
+            this.DehydrateShouldSucceed(new[] { "NewFolder folder successfully dehydrated" }, confirm: true, noStatus: false, foldersToDehydrate: "NewFolder");
+
+            this.Enlistment.UnmountGVFS();
+            fileToCreate.ShouldNotExistOnDisk(this.fileSystem);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                directoryToCreate.ShouldNotExistOnDisk(this.fileSystem);
+            }
+            else
+            {
+                directoryToCreate.ShouldBeADirectory(this.fileSystem);
+            }
+        }
+
+        private void PlaceholderShouldContain(params string[] paths)
+        {
+            string[] placeholderLines = this.GetPlaceholderDatabaseLines();
+            foreach (string path in paths)
+            {
+                placeholderLines.ShouldContain(x => x.StartsWith(path + GVFSHelpers.PlaceholderFieldDelimiter));
+            }
+        }
+
+        private void PlaceholdersShouldNotContain(params string[] paths)
+        {
+            string[] placeholderLines = this.GetPlaceholderDatabaseLines();
+            foreach (string path in paths)
+            {
+                placeholderLines.ShouldNotContain(x => x.StartsWith(path + Path.DirectorySeparatorChar) || x.Equals(path, StringComparison.OrdinalIgnoreCase));

```

Comment:
ah - here we are explicit with comparison for the `Equals` method - should be explicit for the `StartsWith` as well?